### DISCUSSION TITLE
Use of non-public API

### DISCUSSION THREAD

**Discussion by Mx-Iris** [krzyzanowskim/STTextView#49] [Questions]
My application was submitted for AppStore review (it is open source) and was rejected by Apple's review because it used a non-public API. I located that STContextGetFontSmoothingStyle used CGContextGetFontSmoothingStyle, which Apple defined as a non-public API. Below is the reply from Apple's review:

```
Guideline 2.5.1 - Performance - Software Requirements


Your app uses or references the following non-public or deprecated APIs:

Symbols:
• _CGContextGetFontSmoothingStyle

The use of non-public or deprecated APIs is not permitted on the App Store, as they can lead to a poor user experience should these APIs change and are otherwise not supported on Apple platforms.

```

**Comment by krzyzanowskim** [comment]
thank you for reporting this, I should revert that change :-(

**Comment by krzyzanowskim** [comment]
Addressed in 0.9.3
